openerp/osv/fields.py

```diff
@@ -1007,6 +1007,25 @@ def set(self, cr, model, id, name, values, user=None, context=None):
             return
         rel, id1, id2 = self._sql_names(model)
         obj = model.pool[self._obj]
+
+        def insert(ids):
+            # beware of duplicates when inserting
+            query = """ INSERT INTO {rel} ({id1}, {id2})
+                        (SELECT %s, unnest(%s)) EXCEPT (SELECT {id1}, {id2} FROM {rel})
```

Comment:
Shouldn't the EXCEPT be much more restrictive, to avoid performance problems in large m2m tables?
The easiest might be `SELECT %s, unnest(%s)) EXCEPT (SELECT {id1}, {id2} FROM {rel} WHERE {id1} = %s)` (+ pass the extra param in the execute)?